hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -1872,4 +1883,175 @@ public HoodieRecord next() {
       }
     };
   }
+
+  public static HoodieData<HoodieRecord> convertFilesToPartitionStatsRecords(HoodieEngineContext engineContext,
+                                                                             List<DirectoryInfo> partitionInfoList,
```

Comment:
Does this contain log files in MOR tables?

Reply:
No, the assumptions of colstats apply for parition stats too. Log files are considered in subsequent updates with the write status. So, just like colstats, we will have to compact the table before enabling partition stats.

Reply:
I see. Is this for initializing the partition stats only?  Do we have a guard to throw an exception if the colstats or partition stats partition is initialized on the latest file slices with log files?

Reply:
Yes, this is just for initialization. We just log a warning and continue for other filegroups. This is what colstats is also doing. However, given that partition stats will be enabled by default, I think it makes sense to throw exception. Will do it.